protocol/src/main/scala/org/alephium/protocol/vm/Context.scala

```diff
@@ -130,10 +133,23 @@ trait StatelessContext extends CostStrategy {
     indexRaw.v.toInt.flatMap(txEnv.prevOutputs.get).toRight(Right(InvalidTxInputIndex))
   }
 
-  def getTxCaller(indexRaw: Val.U256): ExeResult[Val.Address] = {
+  def getTxInputAddressAt(indexRaw: Val.U256): ExeResult[Val.Address] = {
     getTxPrevOutput(indexRaw).map(output => Val.Address(output.lockupScript))
   }
 
+  def getUniqueTxInputAddress(): ExeResult[Val.Address] = {
+    txEnv.prevOutputs.headOption match {
```

Comment:
Don't you think the call should halt the whole execution if more than one TxInput is present? This will succeed if there are multiple...

Reply:
This will succeed only if the multiple inputs have the same addresses, i.e. unique address

Reply:
I guess I misunderstand the code then. That explanation of behavior is just as good, actually better, than what I expect from the original :)